### PR TITLE
Update Pipeline Service documentation

### PR DESCRIPTION
**Pull request by Roming22**
Add information about the new appstudio-pipeline service account.

### CODE REVIEW

File: book/pipeline-service.md [hunk 1]
```diff
@@ -2,26 +2,36 @@
...
+
+1. _OpenShift Pipelines (OSP):_ It enables the creation and management of Pipelines for user [Applications](https://redhat-appstudio.github.io/book/ref/application-environment-api.html#application).
 2. _Pipelines as Code (PaC):_ An OpenShift Pipelines component that enables receiving webhook events from version control systems. The user would have to install a [GitHub App](https://pipelinesascode.com/docs/install/github_apps/) with the required permissions in order to forward events to the PaC endpoint exposed by the platform.
 3. _Spray Proxy:_ Forwards requests to all clusters in the environment once an event is received on PaC. It is a reverse proxy that broadcasts to multiple backends.
 4. _Tekton Results:_ Responsible for storing logs and pruning PipelineRuns. Utilises AWS RDS as the backend database for our production instance.
 5. _Tekton Chains:_ Provides PipelineRun attestation capabilities, ensuring that all PipelineRuns are verified and can be traced back to their source.
 
+### appstudio-pipeline Service Account
+
+TThe service should offer users a service account for running pipelines.
```
Comment (arewm) [inline book/pipeline-service.md:29]:
```suggestion
The service should offer users a service account for running pipelines.
```

Comment (adambkaplan) [inline book/pipeline-service.md]: nit: spelling

```suggestion
The service should offer users a service account for running pipelines.
```

**Review by arewm — CHANGES_REQUESTED** [review]
Looks good overall on a cursory glance!

File: book/pipeline-service.md (hunk 1 above)
Comment (Roming22) [reply book/pipeline-service.md]: Fixed

**Review by arewm — APPROVED** [review]

**Review by adambkaplan — CHANGES_REQUESTED** [review]
Two items:

1. There should be an associated [ADR](https://github.com/redhat-appstudio/book/tree/main/ADR) with this change - we need to document why we moved away from using the out-of-box RBAC from OpenShift Pipelines and the resulting impact on AppStudio.
2. Since this repo captures technical documentation, we should aim to be more specific when it comes to the following:
   1.  What is in the `appstudio-pipelines-scc` ClusterRole (I also don't see the SCC itself documented anywhere here)
   2. What is in the `appstudio-pipelines-runner` ClusterRole (perhaps a high level description is sufficient).
   3. How CRT creates the associated ClusterRole, service account, and role bindings.

File: book/pipeline-service.md [hunk 2]
```diff
@@ -2,26 +2,36 @@
...
 4. _Tekton Results:_ Responsible for storing logs and pruning PipelineRuns. Utilises AWS RDS as the backend database for our production instance.
 5. _Tekton Chains:_ Provides PipelineRun attestation capabilities, ensuring that all PipelineRuns are verified and can be traced back to their source.
 
+### appstudio-pipeline Service Account
+
+The service should offer users a service account for running pipelines.
+However, the automatic generation of a 'pipeline' service account within namespaces has been disabled in the component because it was found that the permissions granted to that account were overly broad.
+
+The [CodeReadyToolchain](https://github.com/codeready-toolchain) platform (CRT) is in charge of creating an `appstudio-pipeline` account on every tenant namespace, and granting the appropriate RBACs.
+The Pipeline Service component is in charge of the `appstudio-pipelines-scc` ClusterRole, while CRT is responsible for the `appstudio-pipelines-runner` ClusteRole.
```
[the remark this answers is not in the document]
Comment (Roming22) [reply book/pipeline-service.md]: Fixed.

File: book/pipeline-service.md (hunk 1 above)
Comment (Roming22) [reply book/pipeline-service.md]: Fixed

**Comment by Roming22** [comment]
@adambkaplan 
1. ADR added.
2. I've put a link to the definitions of the ClusterRoles. That should be enough, unless some specific permission needs additional context. As for how CRT creates the resources, I'm not sure that it would fall under the AppStudio doc. Is there specific information you had in mind?

File: ADR/0025-appstudio-pipeline-serviceaccount.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
+# 2%. appstudio-pipeline Service Account
```
Comment (adambkaplan) [inline ADR/0025-appstudio-pipeline-serviceaccount.md:1]: Meant to be 25?

```suggestion
# 25. appstudio-pipeline Service Account
```

File: ADR/0025-appstudio-pipeline-serviceaccount.md [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+### ClusterRoles
+
+#### appstudio-pipeline-runner
+
+The resource is defined [here](https://github.com/codeready-toolchain/member-operator/blob/master/config/appstudio-pipelines-runner/base/appstudio_pipelines_runner_role.yaml).
+
+#### appstudio-pipeline-scc
+
+The resource is defined [here](https://github.com/openshift-pipelines/pipeline-service/blob/main/operator/gitops/argocd/pipeline-service/openshift-pipelines/appstudio-pipelines-scc.yaml).
```
Comment (adambkaplan) [inline ADR/0025-appstudio-pipeline-serviceaccount.md:36]: Big :+1: here!

**Review by adambkaplan — APPROVED** [review]
LGTM

File: ADR/0025-appstudio-pipeline-serviceaccount.md (hunk 4 above)
Comment (sbose78) [reply ADR/0025-appstudio-pipeline-serviceaccount.md:36]: 👍

**Pull request merged**